packages/kit/src/core/sync/write_types.js

```diff
@@ -42,24 +48,36 @@ export function write_types(config, manifest_data) {
 				const content = fs.readFileSync(route.page.module, 'utf8');
 				const proxy = tweak_types(content, module_names);
 
-				if (proxy) {
-					// TODO only do this if the module does in fact export `load`
-					write(`${outdir}/proxy${path.basename(route.page.module)}`, proxy);
-					imports.push(`import { load } from './proxy+page.js';`);
+				if (proxy && proxy.exports.includes('load')) {
+					const basename = path.basename(route.page.module);
+					write(`${outdir}/proxy${basename}`, proxy.code);
+					imports.push(`import { load } from './proxy${basename}';`);
 					exports.push(`export type Data = Awaited<ReturnType<typeof load>>;`);
 				} else {
-					// TODO bail out write file with `any` types
+					exports.push(`export type Data = any;`);
 				}
 			}
 
 			if (route.page.server) {
-				// TODO
+				const content = fs.readFileSync(route.page.server, 'utf8');
+				const proxy = tweak_types(content, server_names);
+
+				if (proxy && proxy.exports.includes('GET')) { // TODO handle validation errors from POST/PUT/PATCH
+					const basename = path.basename(route.page.server);
+					write(`${outdir}/proxy${basename}`, proxy.code);
+					imports.push(`import { GET } from './proxy${basename}';`);
+					declarations.push(`type ServerData = Awaited<ReturnType<typeof GET>>;`);
+				} else {
+					declarations.push(`type ServerData = any;`);
+				}
+			} else {
+				declarations.push(`type ServerData = null;`);
```

Comment:
`never`?

Reply:
my thinking was that in the case where there's no `+page.server.js`, or it contains no `GET`, `data` would be `null` inside `load`:

```js
// +page.js
export function load({ data }) {
  if (data) {
    console.log('got data from server', data);
  } else {
    console.log('no data coming from the server, we\'re on our own');
  }

  // ...
}
```

I assumed `type ServerData = null` was the way to do that, but is `never` more appropriate in this case?